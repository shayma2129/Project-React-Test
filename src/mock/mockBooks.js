export default [
    {
        id: "1",
        libelle: "Les Misérables, tome 1",
        auteur: "Victor Hugo",
        edition: 1000,
        nbExemplaire: 15000,
        condition:"Noarchived"
      },
      {
        id: "2",
        libelle: "Le Petit Prince",
        auteur: "Antoine de Saint-Exupéry",
        edition: 800,
        nbExemplaire: 25000,
        condition:"Noarchived"
      },
      {
        id: "3",
        libelle: "Le Rouge et le Noir",
        auteur: "Stendhal",
        edition: 200,
        nbExemplaire: 5420,
        condition:"archived"
      },
  ]