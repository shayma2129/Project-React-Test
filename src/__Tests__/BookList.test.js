import React from "react"
import { render ,fireEvent} from "@testing-library/react"
import BookList from "../components/BooksList/BooksList"
import mockBooks from "../mock/mockBooks"
import user from "@testing-library/user-event"

describe("test list of books", () => {
    test("Testing render an array of Book List",()=>{
        const deletemock=jest.fn()
        const updateConditionBookMock=jest.fn()
        const updateBookMock=jest.fn()
        const updateEmpruntMock=jest.fn()
        const booksComponent=render(<BookList books={mockBooks} deleteBook={deletemock}
            updateBook={updateBookMock} updateConditionBook={updateConditionBookMock} updateEmpruntConditionBook={updateEmpruntMock}/>)
        expect(booksComponent).toMatchSnapshot()
    })
    test("Testing the integration between Book and BookList",()=>{
        const deletemock=jest.fn()
        const updateConditionBookMock=jest.fn()
        const updateBookMock=jest.fn()
        const updateEmpruntMock=jest.fn()
        const mockBookExemple=[  {
            id: 1,
            libelle: "Les Mis√©rables, tome 1",
            auteur: "Victor Hugo",
            edition: 1000,
            nbExemplaire: 15000,
            condition:"Noarchived",
            EmpruntCondition:"emprun",
            dateEmprunt:"thur Jul 3 2020 15:22:16 GMT+0100",
            dateRetourLivre:"Mon Jul 18 2020 13:45:59 GMT+0100"
          }]
        const {getByTestId}=render(<BookList books={mockBookExemple} deleteBook={deletemock}
            updateBook={updateBookMock} updateConditionBook={updateConditionBookMock} updateEmpruntConditionBook={updateEmpruntMock} />)
        
        fireEvent.click(getByTestId("deletebtn"))
        const deleteBookBtn=getByTestId("deletebtn")
        user.click(deleteBookBtn)
        expect(deletemock).toHaveBeenCalled()
        expect(deletemock).toHaveBeenCalledTimes(2)
        
        const AlertforArchive = window.alert;  // remember the jsdom alert
        window.alert = () => {};  // provide an empty implementation for window.alert
        fireEvent.click(getByTestId("archivedbtn"))
        const archivedBookBtn=getByTestId("archivedbtn")
        user.click(archivedBookBtn)
        expect(updateConditionBookMock).toHaveBeenCalled()
        expect(updateConditionBookMock).toHaveBeenCalledTimes(2)
        window.alert = AlertforArchive;

        const jsdomAlert = window.alert;  // remember the jsdom alert
        window.alert = () => {};  // provide an empty implementation for window.alert
        fireEvent.click(getByTestId("Empruntbtn"))
        const empruntBookBtn=getByTestId("Empruntbtn")
        user.click(empruntBookBtn)
        expect(updateEmpruntMock).toHaveBeenCalled()
        expect(updateEmpruntMock).toHaveBeenCalledTimes(2)
        window.alert = jsdomAlert;
        
       /* fireEvent.click(getByTestId("updateBtn"))
        const updateBookBtn=getByTestId("updateBtn")
        user.click(updateBookBtn)
        expect(updateBookMock).toHaveBeenCalled()
        //expect(updateBookMock).toHaveBeenCalledTimes(2)*/

    })
   


})