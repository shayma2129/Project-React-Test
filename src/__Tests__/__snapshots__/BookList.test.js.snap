// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test list of books Testing render an array of Book List 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="BooksList"
      >
        <div
          class="w3-container"
        >
          <div
            class="w3-panel w3-card"
          >
            <div
              class="details-container"
            >
              <h6>
                <ul>
                  <li
                    data-testid="libelle"
                  >
                    Libelle : 
                    Les Misérables, tome 1
                     __
                  </li>
                  <li>
                    Nombre : 
                    15000
                    ___
                  </li>
                  <li>
                    Etat : 
                    Noarchived
                    ___
                  </li>
                  <li>
                    Emprunt : 
                  </li>
                  <div />
                </ul>
              </h6>
            </div>
            <div
              class="actions"
            >
              <div
                aria-label="Second group"
                class="mr-2 btn-group"
                role="group"
              >
                <button
                  class="btn btn-primary"
                  data-testid="ShowModalBtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-eye-open"
                  />
                </button>
                <button
                  class="btn btn-success"
                  data-testid="updateBtn"
                  type="button"
                >
                  <i
                    class=" glyphicon glyphicon-edit"
                  />
                </button>
                <button
                  class="btn btn-danger"
                  data-testid="deletebtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-trash"
                  />
                </button>
                <button
                  class="btn btn-warning"
                  data-testid="archivedbtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-paste"
                  />
                </button>
                <button
                  class="emprunt btn btn-primary"
                  data-testid="Empruntbtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-upload"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div>
             
          </div>
        </div>
        <div
          class="w3-container"
        >
          <div
            class="w3-panel w3-card"
          >
            <div
              class="details-container"
            >
              <h6>
                <ul>
                  <li
                    data-testid="libelle"
                  >
                    Libelle : 
                    Le Petit Prince
                     __
                  </li>
                  <li>
                    Nombre : 
                    25000
                    ___
                  </li>
                  <li>
                    Etat : 
                    Noarchived
                    ___
                  </li>
                  <li>
                    Emprunt : 
                  </li>
                  <div />
                </ul>
              </h6>
            </div>
            <div
              class="actions"
            >
              <div
                aria-label="Second group"
                class="mr-2 btn-group"
                role="group"
              >
                <button
                  class="btn btn-primary"
                  data-testid="ShowModalBtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-eye-open"
                  />
                </button>
                <button
                  class="btn btn-success"
                  data-testid="updateBtn"
                  type="button"
                >
                  <i
                    class=" glyphicon glyphicon-edit"
                  />
                </button>
                <button
                  class="btn btn-danger"
                  data-testid="deletebtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-trash"
                  />
                </button>
                <button
                  class="btn btn-warning"
                  data-testid="archivedbtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-paste"
                  />
                </button>
                <button
                  class="emprunt btn btn-primary"
                  data-testid="Empruntbtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-upload"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div>
             
          </div>
        </div>
        <div
          class="w3-container"
        >
          <div
            class="w3-panel w3-card"
          >
            <div
              class="details-container"
            >
              <h6>
                <ul>
                  <li
                    data-testid="libelle"
                  >
                    Libelle : 
                    Le Rouge et le Noir
                     __
                  </li>
                  <li>
                    Nombre : 
                    5420
                    ___
                  </li>
                  <li>
                    Etat : 
                    archived
                    ___
                  </li>
                  <li>
                    Emprunt : 
                  </li>
                  <div />
                </ul>
              </h6>
            </div>
            <div
              class="actions"
            >
              <div
                aria-label="Second group"
                class="mr-2 btn-group"
                role="group"
              >
                <button
                  class="btn btn-primary"
                  data-testid="ShowModalBtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-eye-open"
                  />
                </button>
                <button
                  class="btn btn-success"
                  data-testid="updateBtn"
                  type="button"
                >
                  <i
                    class=" glyphicon glyphicon-edit"
                  />
                </button>
                <button
                  class="btn btn-danger"
                  data-testid="deletebtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-trash"
                  />
                </button>
                <button
                  class="btn btn-warning"
                  data-testid="archivedbtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-paste"
                  />
                </button>
                <button
                  class="emprunt btn btn-primary"
                  data-testid="Empruntbtn"
                  type="button"
                >
                  <i
                    class="glyphicon glyphicon-upload"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div>
             
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="BooksList"
    >
      <div
        class="w3-container"
      >
        <div
          class="w3-panel w3-card"
        >
          <div
            class="details-container"
          >
            <h6>
              <ul>
                <li
                  data-testid="libelle"
                >
                  Libelle : 
                  Les Misérables, tome 1
                   __
                </li>
                <li>
                  Nombre : 
                  15000
                  ___
                </li>
                <li>
                  Etat : 
                  Noarchived
                  ___
                </li>
                <li>
                  Emprunt : 
                </li>
                <div />
              </ul>
            </h6>
          </div>
          <div
            class="actions"
          >
            <div
              aria-label="Second group"
              class="mr-2 btn-group"
              role="group"
            >
              <button
                class="btn btn-primary"
                data-testid="ShowModalBtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-eye-open"
                />
              </button>
              <button
                class="btn btn-success"
                data-testid="updateBtn"
                type="button"
              >
                <i
                  class=" glyphicon glyphicon-edit"
                />
              </button>
              <button
                class="btn btn-danger"
                data-testid="deletebtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-trash"
                />
              </button>
              <button
                class="btn btn-warning"
                data-testid="archivedbtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-paste"
                />
              </button>
              <button
                class="emprunt btn btn-primary"
                data-testid="Empruntbtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-upload"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div>
           
        </div>
      </div>
      <div
        class="w3-container"
      >
        <div
          class="w3-panel w3-card"
        >
          <div
            class="details-container"
          >
            <h6>
              <ul>
                <li
                  data-testid="libelle"
                >
                  Libelle : 
                  Le Petit Prince
                   __
                </li>
                <li>
                  Nombre : 
                  25000
                  ___
                </li>
                <li>
                  Etat : 
                  Noarchived
                  ___
                </li>
                <li>
                  Emprunt : 
                </li>
                <div />
              </ul>
            </h6>
          </div>
          <div
            class="actions"
          >
            <div
              aria-label="Second group"
              class="mr-2 btn-group"
              role="group"
            >
              <button
                class="btn btn-primary"
                data-testid="ShowModalBtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-eye-open"
                />
              </button>
              <button
                class="btn btn-success"
                data-testid="updateBtn"
                type="button"
              >
                <i
                  class=" glyphicon glyphicon-edit"
                />
              </button>
              <button
                class="btn btn-danger"
                data-testid="deletebtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-trash"
                />
              </button>
              <button
                class="btn btn-warning"
                data-testid="archivedbtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-paste"
                />
              </button>
              <button
                class="emprunt btn btn-primary"
                data-testid="Empruntbtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-upload"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div>
           
        </div>
      </div>
      <div
        class="w3-container"
      >
        <div
          class="w3-panel w3-card"
        >
          <div
            class="details-container"
          >
            <h6>
              <ul>
                <li
                  data-testid="libelle"
                >
                  Libelle : 
                  Le Rouge et le Noir
                   __
                </li>
                <li>
                  Nombre : 
                  5420
                  ___
                </li>
                <li>
                  Etat : 
                  archived
                  ___
                </li>
                <li>
                  Emprunt : 
                </li>
                <div />
              </ul>
            </h6>
          </div>
          <div
            class="actions"
          >
            <div
              aria-label="Second group"
              class="mr-2 btn-group"
              role="group"
            >
              <button
                class="btn btn-primary"
                data-testid="ShowModalBtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-eye-open"
                />
              </button>
              <button
                class="btn btn-success"
                data-testid="updateBtn"
                type="button"
              >
                <i
                  class=" glyphicon glyphicon-edit"
                />
              </button>
              <button
                class="btn btn-danger"
                data-testid="deletebtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-trash"
                />
              </button>
              <button
                class="btn btn-warning"
                data-testid="archivedbtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-paste"
                />
              </button>
              <button
                class="emprunt btn btn-primary"
                data-testid="Empruntbtn"
                type="button"
              >
                <i
                  class="glyphicon glyphicon-upload"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div>
           
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
